packages/react-core/src/components/TreeView/TreeViewList.tsx

```diff
@@ -18,11 +23,35 @@ export const TreeViewList: React.FunctionComponent<TreeViewListProps> = ({
   isNested = false,
   onSearch,
   searchProps,
+  toolbarItems,
+  toolbarLeft,
   children,
   ...props
 }: TreeViewListProps) => (
   <>
-    {onSearch && (
+    {toolbarItems && onSearch && toolbarLeft && (
+      <React.Fragment>
+        <Toolbar>
+          <ToolbarContent>
+            {toolbarItems}
+            <TreeViewSearch onChange={onSearch} {...searchProps} />
+          </ToolbarContent>
+        </Toolbar>
+        <Divider />
+      </React.Fragment>
+    )}
+    {toolbarItems && onSearch && !toolbarLeft && (
+      <React.Fragment>
+        <Toolbar>
+          <ToolbarContent>
+            <TreeViewSearch onChange={onSearch} {...searchProps} />
+            {toolbarItems}
+          </ToolbarContent>
+        </Toolbar>
+        <Divider />
+      </React.Fragment>
+    )}
+    {!toolbarItems && onSearch && (
```

Comment:
@gildub Instead of having one `toolbarItems` prop and a boolean for whether to put it on the left or right, it might be easier to have `toolbarItemsLeft` and `toolbarItemsRight` props. That way we can support having items on both the left and the right, and we don't need two conditions here.

We also need to support the case where we have toolbar items but no search bar.